assets/js/src/core/modules/asset/editor/types/video/tab-manager/tabs/preview/preview-container.tsx

```diff
@@ -11,22 +11,57 @@
 *  @license    https://github.com/pimcore/studio-ui-bundle/blob/1.x/LICENSE.md POCL and PCL
 */
 
-import React, { useContext } from 'react'
+import React, { createContext, useContext, useMemo, useState } from 'react'
 import { PreviewView } from './preview-view'
 import { useGetAssetByIdQuery } from '@Pimcore/modules/asset/asset-api-slice.gen'
 import { AssetContext } from '@Pimcore/modules/asset/asset-provider'
 import { ContentToolbarSidebarView } from '@Pimcore/modules/element/editor/tab-manager/layouts/content-toolbar-sidebar-view'
+import { sidebarManager } from '@Pimcore/modules/asset/editor/types/video/tab-manager/tabs/preview/sidebar'
+import { Sidebar } from '@Pimcore/components/sidebar/sidebar'
+
+export interface IVideoContext {
+  thumbnail: string
+  setThumbnail: React.Dispatch<React.SetStateAction<string>>
+  playerPosition: number
+  setPlayerPosition: React.Dispatch<React.SetStateAction<number>>
+
+}
+
+export const VideoContext =
+  createContext<IVideoContext>({
+    thumbnail: '',
+    setThumbnail: () => {},
+    playerPosition: 0,
+    setPlayerPosition: () => {}
+  })
 
 const PreviewContainer = (): React.JSX.Element => {
+  const [thumbnail, setThumbnail] = useState<string>('')
+  const [playerPosition, setPlayerPosition] = useState<number>(0)
   const assetContext = useContext(AssetContext)
   const { data } = useGetAssetByIdQuery({ id: assetContext.id! })
+  const sidebarEntries = sidebarManager.getEntries()
+  const sidebarButtons = sidebarManager.getButtons()
+
+  const contextValue = useMemo<IVideoContext>(() => ({
+    thumbnail,
+    setThumbnail,
+    playerPosition,
+    setPlayerPosition
+  }), [thumbnail, playerPosition])
 
   return (
-    <ContentToolbarSidebarView>
-      <PreviewView
-        src={ data!.fullPath! }
-      />
-    </ContentToolbarSidebarView>
+    <VideoContext.Provider value={ contextValue }>
```

Comment:
We could consider to move all context related code to a video provider component. That makes it a bit easier to reuse. :)

So far, when I interduce a new Context. I normally create a new Provider and a custom hook e.g. `useVideo`.
And it seems quite useful. In the hook you can easily switch to a new context in case we switch to another provider. And having the Provider abstracted makes it of course super reusable. :) 

What do you think?